File: Assignment_2/Q1a.py

#--------------------* main function to run some test *------------------------
def main():
    points1 = [1,2,3,4,5,6]  # A list variable to store list of points.
    print("Average distance between points [1,2,3,4,5,6] is " ,averageDistance(points1))
    points2 = [3,4,5,6,7,8]  # another list of points.
    print("Average distance between points [3,4,5,6,7,8] is " ,averageDistance(points2))
    


#-------------------* arvearge distance function *---------------------------
# this function take a array of 6 points and return average distance.
def averageDistance(numArray):
    sum =0.0    # variable to store sum of distance between points.
    for i in range(0,6): # fixing one point.    
        for j in range(0,6): # runing over the points. 
            sum += abs(numArray[i] - numArray[j]) # adding abs value of difference between points.
    return (sum/36) # returning the average by dividing 36.


# -----------------------* Output of this code *-----------------------------
#('Average distance between points [1,2,3,4,5,6] is ', 1.9444444444444444)
#('Average distance between points [3,4,5,6,7,8] is ', 1.9444444444444444)




















if __name__ == "__main__":
    main()

File: Assignment_2/Q1a.cpp
#include <iostream>  // cin , cout 
#include <cmath> // abs(),

using namespace std;

double averageDistance(double* );

//-------------------------* main function to run tests *-------------------------
int main()
{
    double points1[6] = {1,2,3,4,5,6};    // double[] varible to store the points with unit distance 
    cout<<"Average distance of points [ 1 , 2, 3 , 4, 5 ,6 ] is :" << averageDistance(points1)<<endl;

    double points2[6] = {3,4,5,6,7,8};   // double[] variable store points with unit distance (displaced by 2 unit)
    cout<<"Average distance of points [ 3 , 4, 5 ,6 , 7, 8 ] is :" << averageDistance(points2)<<endl;
}

//-------------------------* averageDistance funtion to find the averageDistance between 6 points *-------------------------
double  averageDistance(double a[])
{
    double sum=0;           //  Variable to store the sum during loop
    for(int i=0;i<6;i++)    //  fixing one of the points out of six
    for(int j=0;j<6;j++)    //  Running over the rest of the points including fixed point
    {
        sum+= abs(a[j]-a[i]);    // Adding the absolute distance between the points.
    }
    return (sum/36);         // returing the average distance of points.
}

//------------------------* Output of the code *--------------------------------------------
//Average distance of points [ 1 , 2, 3 , 4, 5 ,6 ] is :1.94444
//Average distance of points [ 3 , 4, 5 ,6 , 7, 8 ] is :1.94444